File: Data Structures & Algorithms/longest-consecutive-sequence/submission-7.java
class Solution {
    public int longestConsecutive(int[] nums) {
        //checking 2
        //if (1) 
        //length = length + 1
        //iterate to next num
        //else{
            //length = 1
        if(nums.length == 0){
            return 0;
        }
        Set<Integer> numbers = new HashSet();
        for(int n : nums){
            numbers.add(n);
        }
        //2 20 4 10 3 5
        int max = 0;
        for(int n : nums){
            int length = 1;
            while(numbers.contains(n-1)){
                length = length + 1;
                n = n-1;
            }
            if(length >= max){
                max = length;
            }
        }

        return max;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-7.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l, r = 0, len(nums) - 1
        #1 2 3 4 5 6
        #6 1 2 3 4 5
        #5 6 1 2 3 4
        #4 5 6 1 2 3
        #3 4 5 6 1 2
        #2 3 4 5 6 1
        while l <= r:
            m = math.floor((r+l)/2)
            if nums[m] < nums[r]:
                r = m
            elif nums[m] > nums[r]:
                l = m + 1
            else: return nums[m]

        return -1